nym-vpn-lib/src/mixnet_connect.rs

```diff
@@ -289,6 +290,14 @@ pub(crate) async fn setup_mixnet_client(
     debug!("mixnet client has wireguard_mode={enable_wireguard}");
     let mixnet_client = if let Some(path) = mixnet_client_key_storage_path {
         debug!("Using custom key storage path: {:?}", path);
+        let gateway_id = mixnet_entry_gateway.to_base58_string();
+        if let Err(err) =
+            credentials::check_imported_credential(path.to_path_buf(), &gateway_id).await
+        {
+            // UGLY: flow needs to restructured to sort this out, but I don't want to refactor all that just before release.
+            task_client.mark_as_success();
```

Comment:
`.disarm()` would be more descriptive ...

Reply:
then update `nym` dependency version and call it : )